### PR TITLE
Introduce Api tokens support

### PR DESCRIPTION
**Pull request by dataclouder**
Add methods `VCDClient.GetBearerTokenFromApiToken` and modifies `VCDClient.SetToken` to accept API tokens transparently.
(Available in VCD 10.3.1)

### CODE REVIEW

File: govcd/api_token.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	var body []byte
+	var tokenDef types.ApiTokenRefresh
+	if resp.Body != nil {
+		body, err = ioutil.ReadAll(resp.Body)
+	}
+	responseData := "[" + strings.Repeat("*", 10) + "]"
```
Comment (lvirbalas) [inline govcd/api_token.go:74]: Please add a comment why are we doing this.

Comment (dataclouder) [reply govcd/api_token.go:76]: This is an obfuscation of sensitive data for logging. 
Comment added

**Review by lvirbalas — APPROVED** [review]

File: util/logging.go [hunk 2]
```diff
@@ -160,34 +160,34 @@ func SetLog() {
...
-	out = re1.ReplaceAllString(in, `${1}******${3}`)
+	re3 := regexp.MustCompile(`(.*<e:CipherValue>)(.*)(</e:CipherValue>.*)`)
+	out = re3.ReplaceAllString(out, `${1}******${3}`)
 	// Token data between <xenc:CipherValue> </xenc:CipherValue>
-	re2 := regexp.MustCompile(`(.*<xenc:CipherValue>)(.*)(</xenc:CipherValue>.*)`)
-	out = re2.ReplaceAllString(out, `${1}******${3}`)
+	re4 := regexp.MustCompile(`(.*<xenc:CipherValue>)(.*)(</xenc:CipherValue>.*)`)
+	out = re4.ReplaceAllString(out, `${1}******${3}`)
+
+	// Data inside certificates and private keys
```
Comment (vbauzysvmware) [inline util/logging.go:186]: Thank you

**Review by vbauzysvmware — APPROVED** [review]
LGTM

File: govcd/api_token.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+)
+
+// SetApiToken behaves similarly to SetToken, with the difference that it will
+// return full information about the bearer token, so that the caller can make decisions about token expiration
+func (vcdCli *VCDClient) SetApiToken(org, apiToken string) (*types.ApiTokenRefresh, error) {
+	usableApiToken, err := vcdCli.GetBearerTokenFromApiToken(org, apiToken)
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/api_token.go]: changer to refreshToken, as the value is not a bearer token

File: govcd/api_token.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		return nil, err
+	}
+	err = vcdCli.SetToken(org, BearerTokenHeader, usableApiToken.AccessToken)
+	if err != nil {
+		return nil, err
+	}
+	return usableApiToken, nil
+}
+
+// GetBearerTokenFromApiToken receives an API token and retrieves a bearer token
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/api_token.go]: changed

**Review by Didainius — APPROVED** [review]

**Pull request merged**